### PR TITLE
chore: expose Anvil URL from GitHub actions

### LINKED ISSUE #526: Expose RPC URL from github action runner
Depends on: https://github.com/ubiquity/ubiquity-dollar/pull/417

What should be done:
1. When https://github.com/ubiquity/ubiquity-dollar/pull/417 is ready we should deploy contracts in their initial state when ["deploy test blockchain" workflow](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/deploy-anvil.yml) starts. The "deploy test blockchain" workflow can be started only manually.
2. When user runs the "deploy test blockchain" workflow it should print the deployed RPC URL in the workflow logs so that everybody could connect to that test node.

### PR DESCRIPTION
**Pull request by gitcoindev**
Resolves: #526

QA: https://github.com/gitcoindev/ubiquity-dollar/issues/1

### CODE REVIEW

File: .github/workflows/setup-anvil.yml [hunk 1]
```diff
@@ -19,8 +19,22 @@ jobs:
           version: "latest"
       - name: Tunnel Anvil
         id: tunnel
-        run: cloudflared tunnel --url http://localhost:8545 &
+        run: cloudflared tunnel --url http://localhost:8545 --logfile cloudflared.log &
         shell: bash
+      - name: Extract Cloudflared URL
+        run: grep -E "https://.*.trycloudflare.com" cloudflared.log | cut -d'|' -f2 | tr -d ' ' > cloudflare-url.txt
```
Comment (rndquu) [inline .github/workflows/setup-anvil.yml]: We should print the extracted URL to the logs

**Review by rndquu — CHANGES_REQUESTED** [review]
@gitcoindev Did you test how long the tunnel is alive?

File: .github/workflows/setup-anvil.yml (hunk 1 above)
Comment (0x4007) [inline .github/workflows/setup-anvil.yml]:
```suggestion
        run: grep -E "https://.*.trycloudflare.com" cloudflared.log | cut -d'|' -f2 | tr -d ' ' > cloudflare-url.txt
        cat cloudflare-url.txt
```

Does that work?

**Comment by gitcoindev** [comment]
> @gitcoindev Did you test how long the tunnel is alive?

@rndquu yes, it was alive for 6 hours. I also connected 5 hours after the GitHub action started and Anvil was still responding to my calls.

```
[Setup Anvil](https://github.com/gitcoindev/ubiquity-dollar/actions/runs/5880082791)
Setup Anvil #5: Manually run by [gitcoindev](https://github.com/gitcoindev)
 yesterday
 6h 0m 23s
```

File: .github/workflows/setup-anvil.yml (hunk 1 above)
Comment (gitcoindev) [reply .github/workflows/setup-anvil.yml]: @pavlovcik this should work, let me quickly try to confirm and then update the PR

**Comment by 0x4007** [comment]
> > @gitcoindev Did you test how long the tunnel is alive?
> 
> @rndquu yes, it was alive for 6 hours. I also connected 5 hours after the GitHub action started and Anvil was still responding to my calls.
> 
> ```
> [Setup Anvil](https://github.com/gitcoindev/ubiquity-dollar/actions/runs/5880082791)
> Setup Anvil #5: Manually run by [gitcoindev](https://github.com/gitcoindev)
>  yesterday
>  6h 0m 23s
> ```

Amazing to see!

File: .github/workflows/setup-anvil.yml (hunk 1 above)
Comment (gitcoindev) [reply .github/workflows/setup-anvil.yml]: @pavlovcik yes , it does work, I updated the PR. URL was also visible in my fork's GitHub action log: 

```
Run grep -E "https://.*.trycloudflare.com" cloudflared.log | cut -d'|' -f2 | tr -d ' ' > cloudflare-url.txt
https://patients-chose-pets-registration.trycloudflare.com/
```

Comment (gitcoindev) [reply .github/workflows/setup-anvil.yml]: done

File: .github/workflows/setup-anvil.yml [hunk 2]
```diff
@@ -19,8 +19,24 @@ jobs:
...
         id: tunnel
-        run: cloudflared tunnel --url http://localhost:8545 &
+        run: cloudflared tunnel --url http://localhost:8545 --logfile cloudflared.log &
         shell: bash
+      - name: Extract Cloudflared URL
+        run: |
+          grep -E "https://.*.trycloudflare.com" cloudflared.log | cut -d'|' -f2 | tr -d ' ' > cloudflare-url.txt
+          cat cloudflare-url.txt
+        shell: bash
+      - name: Store URL
```
Comment (rndquu) [inline .github/workflows/setup-anvil.yml]: Do we still need this step? Along with "Create comment"

Comment (gitcoindev) [reply .github/workflows/setup-anvil.yml]: @rndquu it is up to you and @pavlovcik to decide. I added this to prove my QA, create an artifact and add comment to JIRA, can safely remove both now. Please let me know and I will do this asap if needed.

Comment (rndquu) [reply .github/workflows/setup-anvil.yml]: Pls remove

Comment (gitcoindev) [reply .github/workflows/setup-anvil.yml]: done

**Review by rndquu — APPROVED** [review]

**Review by 0x4007 — APPROVED** [review]

**Comment by gitcoindev** [comment]
@rndquu @pavlovcik thank you for approving and merging!

**Comment by 0x4007** [comment]
@gitcoindev thanks for shipping!

**Pull request merged**